test/e2e/instrumentation/monitoring/custom_metrics_autoscaling.go

```diff
@@ -0,0 +1,205 @@
+
+	// Autoscale the deployments
+	err = createPodsHPA(kubeClient)
+	if err != nil {
+		framework.Failf("Failed to create 'Pods' HPA: %v", err)
+	}
+	err = createObjectHPA(kubeClient)
+	if err != nil {
+		framework.Failf("Failed to create 'Objects' HPA: %v", err)
+	}
+
+	// Wait a for HPA to scale down targets
+	// TODO: wait for the deployments to reach the expected numbers of replicas instead of fixed
+	//       amount of time
+	time.Sleep(240 * time.Second)
+
+	// Verify that the deployments were scaled down to the minimum value
+	sdExporterDeployment, err := kubeClient.Extensions().Deployments("default").Get("stackdriver-exporter-deployment", metav1.GetOptions{})
+	if err != nil {
+		framework.Failf("Failed to retrieve info about 'stackdriver-exporter-deployment'")
+	}
+	if *sdExporterDeployment.Spec.Replicas != 1 {
+		framework.Failf("Unexpected number of replicas for 'stackdriver-exporter-deployment'. Expected 1, but received %v", *sdExporterDeployment.Spec.Replicas)
+	}
+	dummyDeployment, err := kubeClient.Extensions().Deployments("default").Get("dummy-deployment", metav1.GetOptions{})
+	if err != nil {
+		framework.Failf("Failed to retrieve info about 'dummy-deployment'")
+	}
+	if *sdExporterDeployment.Spec.Replicas != 1 {
+		framework.Failf("Unexpected number of replicas for 'dummy-deployment'. Expected 1, but received %v", *dummyDeployment.Spec.Replicas)
+	}
+
+	framework.ExpectNoError(err)
+}
+
+func createDeploymentsToScale(cs clientset.Interface) error {
+	_, err := cs.Extensions().Deployments("default").Create(StackdriverExporterDeployment("stackdriver-exporter-deployment", 2, 100))
+	if err != nil {
+		return err
+	}
+	_, err = cs.Core().Pods("default").Create(StackdriverExporterPod("stackdriver-exporter-pod", "stackdriver-exporter-pod", CustomMetricName, 100))
+	if err != nil {
+		return err
+	}
+	_, err = cs.Extensions().Deployments("default").Create(StackdriverExporterDeployment("dummy-deployment", 2, 100))
+	return err
+}
+
+func cleanupDeploymentsToScale(cs clientset.Interface) {
+	_ = cs.Extensions().Deployments("default").Delete("stackdriver-exporter-deployment", &metav1.DeleteOptions{})
+	_ = cs.Core().Pods("default").Delete("stackdriver-exporter-pod", &metav1.DeleteOptions{})
+	_ = cs.Extensions().Deployments("default").Delete("dummy-deployment", &metav1.DeleteOptions{})
+}
+
+func createPodsHPA(cs clientset.Interface) error {
+	_, err := cs.AutoscalingV2beta1().HorizontalPodAutoscalers("default").Create(&autoscaling.HorizontalPodAutoscaler{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "custom-metrics-pods-hpa",
+			Namespace: "default",
+		},
+		Spec: autoscaling.HorizontalPodAutoscalerSpec{
+			Metrics: []autoscaling.MetricSpec{
+				{
+					Type: autoscaling.PodsMetricSourceType,
+					Pods: &autoscaling.PodsMetricSource{
+						MetricName:         CustomMetricName,
+						TargetAverageValue: *resource.NewQuantity(200, resource.DecimalSI),
+					},
+				},
+			},
+			MaxReplicas: 3, // default min is 1
```

Comment:
Nitpicking: Wouldn't it be better to explicitly set MinReplicas here rather than comment the default happens to be right, if the test relies on it?

Reply:
Done.